tests/multi_ptr/multi_ptr_access_members.h

```diff
@@ -0,0 +1,229 @@
+
+/*******************************************************************************
+//
+//  SYCL 2020 Conformance Test Suite
+//
+//  Provides code for multi_ptr access members
+//
+*******************************************************************************/
+#ifndef __SYCLCTS_TESTS_MULTI_PTR_ACCESS_MEMBERS_OPS_H
+#define __SYCLCTS_TESTS_MULTI_PTR_ACCESS_MEMBERS_OPS_H
+
+#include "../common/common.h"
+
+#include "../common/section_name_builder.h"
+#include "multi_ptr_common.h"
+
+namespace multi_ptr_access_members {
+
+/**
+ * @brief Provides functions for verification multi_ptr access operators that
+ *        they returns correct data
+ * @tparam T Current data type
+ * @tparam AddrSpaceT sycl::access::address_space enumeration's field
+ * @tparam IsDecoratedT sycl::access::decorated enumeration's field
+ */
+template <typename T, typename AddrSpaceT, typename IsDecoratedT>
+class run_access_members_tests {
+  static constexpr sycl::access::address_space space = AddrSpaceT::value;
+  static constexpr sycl::access::decorated decorated = IsDecoratedT::value;
+  using multi_ptr_t = const sycl::multi_ptr<T, space, decorated>;
+
+ public:
+  /**
+   * @param type_name Current data type string representation
+   * @param address_space_name Current sycl::access::address_space string
+   *        representation
+   * @param is_decorated_name Current sycl::access::decorated string
+   *        representation
+   */
+  void operator()(const std::string &type_name,
+                  const std::string &address_space_name,
+                  const std::string &is_decorated_name) {
+    auto queue = sycl_cts::util::get_cts_object::queue();
+    constexpr int val_to_init = 42;
+    T value = user_def_types::get_init_value_helper<T>(val_to_init);
+    // Variables that will be used to check that access members returns correct
+    // data type
+    bool dereference_return_type_is_correct = false;
+    bool dereference_op_return_type_is_correct = false;
+    bool get_return_type_is_correct = false;
+    bool get_raw_return_type_is_correct = false;
+    bool get_decorated_return_type_is_correct = false;
+    // Variables that will be used to check that access members returns correct
+    // value
+    T dereference_ret_value;
+    T dereference_op_ret_value;
+    T get_member_ret_value;
+    T get_raw_member_ret_value;
```

Comment:
Just a random idea. Can we simplify the code by creating a structure with fields matching all these variables?